packages/rest/test/integration/rest.server.integration.ts

```diff
@@ -314,14 +300,35 @@ servers:
         cert: fs.readFileSync(certPath),
       },
     });
-    server.handler(({response}, sequence) => {
-      response.end();
+    server.handler(dummyRequestHandler);
+    await server.start();
+    const serverUrl = server.getSync(RestBindings.URL);
+    const res = await httpsGetAsync(serverUrl);
+    expect(res.statusCode).to.equal(200);
+    await server.stop();
+  });
+
+  it('honors HTTPS config binding after instantiation', async () => {
+    const keyPath = path.join(__dirname, 'key.pem');
+    const certPath = path.join(__dirname, 'cert.pem');
+    const server = await givenAServer({
+      rest: {
+        port: 0,
+        host: '::1',
+        protocol: 'https',
+      },
+    });
+    server.handler(dummyRequestHandler);
+    expect(server.start()).to.be.rejected();
```

Comment:
Missing `await`.

```ts
await expect(server.start()).to.be.rejected();
```

There are few more places in this test file where we are expecting a rejected promise and don't waiting until the expectation is checked, please fix them too.